JoomGalleryfriends/JG4-dev
# Dynamic list of configuration options (dropdown-list)

This PR adds a general implementation for dynamic configuration options dopdown lists. One application is the selection of available filesystem adapters (plugin type filesystem).

## How is it done
In the XML form definition, add the attribute `dynamic="true"` to the field you want to add dynamic options. The field has to be of a type which supports options like e.g `list` or `radio`. Additionally you have to add the attribute `script="scriptName"` where `scriptName` is the name of the script providing the field with available options.

The script to load the available options can be provided in a plugin listening to the `onJoomGetOptions` event or within the `Joomgallery\Component\Joomgallery\Administrator\Helper\ConfigHelper` where scriptName has to be an available method of this helper class.

## How to test this PR
PR affects only the form view of configuration sets.
Configuration sets form view should work the same as before except for the fact that the field called "Filesystem" now provide a dorpdown list with all available filesystem adapters available in your system.
![grafik](https://user-images.githubusercontent.com/39154009/232501879-f53c436c-2f1f-4d32-b953-9cae261e7e90.png)

You can add new filesystems by install new filesystem plugins or by adding new adapters to already installed filesystem plugins editing the plugin parameters.
